libraries/ESP8266WiFi/src/include/ClientContext.h

```diff
@@ -130,8 +130,11 @@ class ClientContext
         }
         _connect_pending = 1;
         _op_start_time = millis();
-        // This delay will be interrupted by esp_schedule in the connect callback
-        delay(_timeout_ms);
+        // Following delay will be interrupted by connect callback
+        for (decltype(_timeout_ms) i = 0; _connect_pending && i < _timeout_ms; i++) {
```

Comment:
This looks like the same logic in polledTimeout. @d-a-v does it make sense to replace usage here?

Reply:
The original `delay()` is supposed to be interrupted by `esp_schedule()` anytime (after the original included `yield()`)
In this PR it is divided into small delays but the desired effect is the same. We want to continue as soon as a packet is received (which calls `esp_schedule()`) without latency.
(afaik PolledTimeout is not interruptible)

Reply:
I was thinking of PolledTimeout with different policies than we currently have. But maybe it's not worth implementing for these cases.